client/src/components/LoginSigninForm.jsx

```diff
@@ -0,0 +1,168 @@
+        if (err.response.data.message) {
+            toast.error(err.response.data.message, 
+                { position: toast.POSITION.TOP_CENTER });
+        }
+    } else if (err.message) {
+        toast.error(err.message, 
+            { position: toast.POSITION.TOP_CENTER });
+    } else {
+        toast.error('Sorry, an error occured, try again later', 
+            { position: toast.POSITION.TOP_CENTER });
+        console.log('Error', err);
+    }
+}
+
+
+export default class LoginSigninFrom extends React.PureComponent {
+    state = {
+        email: '',
+        password: '',
+        username: '',
+        firstName: '',
+        lastName: '',
+        ageCheck: false,
+        formFunctionality: ['Login', 'New user? ', 'Sign-in'],
+        loading: false,
+    }
+
+    handleFormSubmit = async (event) => {
+        event.preventDefault()
+
+        if (this.state.formFunctionality[0] === 'Login') {
+            const user = {
+                email: this.state.email,
+                password: this.state.password
+            }
+            try {
+                this.setState({loading: true})
+                const { data } = await axios.patch('http://localhost:3129/users/login', user)
+
+                this.setState({ loading: false })
+                this.props.formSubmit(data.payload, this.state.password)
+            } catch (err) {
+                this.setState({loading: false})
+                handleNetworkErrors(err)
+            }
+        } else if (this.state.formFunctionality[0] === 'Sign-in') {
+            const user = { 
+                username: this.state.username, 
+                firstname: this.state.firstName, 
+                lastname: this.state.lastName,
+                password: this.state.password, 
+                email: this.state.email
+            }
+            if (this.state.ageCheck) {
+                user.ageCheck = 'true'
+            }
+            try {
+                this.setState({loading: true})
+                const { data } = await axios.post('http://localhost:3129/users/signup', user)
+
+                this.setState({ loading: false })
+                this.props.formSubmit(data.payload, this.state.password)
+            } catch (err) {
+                this.setState({loading: false})
+                handleNetworkErrors(err)
+            }
+        }
+    }
+
+    handleUsernameInput = event => {
+        this.setState({username: event.target.value})
+    }
+
+
+    handleFirstNameInput = event => {
+        this.setState({firstName: event.target.value})
+    }
+    
+    handleLastNameInput = event => {
+        this.setState({lastName: event.target.value})
+    }
+    
+    handleDobInput = event => {
+        this.setState({dob: event.target.value})
+    }
+    
+    handleEmailInput = event => {
+        this.setState({email: event.target.value})
+    }
+
+    handlePasswordInput = event => {
+        this.setState({password: event.target.value})
+    }
+
```

Comment:
i think you could use onchange and event bubbing to combine most if not all of these into one handler